src/main/java/net/spy/memcached/MemcachedConnection.java

```diff
@@ -373,7 +374,25 @@ public MemcachedConnection(final int bufSize, final ConnectionFactory f,
     setDaemon(f.isDaemon());
     start();
   }
-  
+
+  private void waitForNodeUpdateCondition() {
+    conditionLock.lock();
+    try{
+      try {
+        if (isNodeUpdateConditionInClientInit) {
+          setNodeUpdateConditionInClientInit(false);
+        }
+        nodeUpdateCondition.await(50L, TimeUnit.SECONDS);
+      } catch (InterruptedException e) { }
+    } finally{
+      conditionLock.unlock();
+    }
+  }
+
+  public void setNodeUpdateConditionInClientInit(boolean status) {
+    isNodeUpdateConditionInClientInit = status;
+  }
```

Comment:
This doesn't make sense to me. Here we need to solve 2 separate problems and let's not mix them together like this. 

1. Synchronization between the PollerThread and the MemcachedConnection thread, which the current logic is fine as is. 
2. Synchronization between the MemcachedClient constructor thread and the MemcachedConnection thread for the case when the constructor thread discovered the cluster configuration and needs to wait for the MemcachedConnection thread to apply it before returning to the user. I would create a separate routine to do that.